Python-packages/covidcast-py/covidcast/plotting.py

```diff
@@ -170,6 +172,29 @@ def get_geo_df(data: pd.DataFrame,
     return output
 
 
+def animate(data: pd.DataFrame, filepath: str, fps: int = 3, dpi: int = 150, **kwargs) -> None:
+    """Generate an animated mp4 file of a signal over time.
+
+    :param data: DataFrame for a single signal over time.
+    :param filepath: Path where video will be saved.
+    :param fps: Frame rate in frames per second for animation. Defaults to 3.
+    :param dpi: Dots per inch for output video. Defaults to 150 on a 12.8x9.6 figure (1920x1440).
+    :return: None
+    """
+    # probesize is set to avoid warning by ffmpeg on frame rate up to 4k resolution.
+    writer = imageio.get_writer(filepath, fps=fps, input_params=["-probesize", "75M"])
+    num_days = (max(data.time_value) - min(data.time_value)).days
+    day_list = [min(data.time_value) + timedelta(days=x) for x in range(num_days+1)]
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        for d in day_list:
+            print(f"Processing {d.date()}")
+            plot_choropleth(data, time_value=d, **kwargs)
+            plt.savefig(f"{tmpdirname}/{d}.jpg", format="jpg", dpi=dpi)
```

Comment:
You could probably use `io.BytesIO` to save the figure to memory, then `imread` from that, so no temp files needed... maybe?

Also, is jpg the best option here, since it ensures each frame will have compression artifacts even before mp4 compression? Maybe png

Reply:
Ok this is weird. I swear two nights ago when I tried this with png it was giving me the weird bold text issue (even on mp4, which didn't make sense) but it's to be working now.

Good call on the bytesio, just added that